pkg/client/client.go

```diff
@@ -928,55 +929,53 @@ func (c Client) validatePrometheusResource(ctx context.Context, prom types.Names
 
 	avail, err := getMonitoringCondition(p.Status.Conditions, monv1.Available)
 	if err != nil {
-		err = fmt.Errorf("prometheus: %w", err)
-		// failing to get Prometheus.Status.Condtion -> Degraded: Unknown & Unavailable: Unknown
+		// Report Degraded=Unknown and Unavailable=Unknown if the condition can't be found.
```

Comment:
/hold
Sorry...did you mean to remove the `Errorf` here?

Reply:
I'm leaning towards adding more context, sth like
`"Prometheus CR %s: %w", prom.String()...` to make sure users will go take a look at the right CR status for more details.
I was also testing the inclusion of the CR reason/message directly to the error message https://github.com/openshift/cluster-monitoring-operator/pull/2405/files

Reply:
yes the removal was intentional.  The returned error is already informative (it says "failed to find condition ...").
I'm updating the patch to always add the resource identifier.